### ISSUE TITLE
Assignment 2 Question

### ISSUE THREAD

**Issue by sma4396** [hil-se/fds#79]
Working on Assignment 2. How should we handle a situation where in the training set a certain attribute is the same across the entire column (say 't'), but in the test set it uses another attribute (say 'f')?

**Comment by azhe825** [comment]
@sma4396 
This is a great question:
1. We cannot do anything for this in training (since we do not know there is another possible value 'f' for that attribute).
2. We can handle this in prediction: treat that similarly as a missing value for the test data. The my_NB_hint.py file actually resolves this issue by [if value in self.P[label][key] else 1](https://github.com/hil-se/fds/blob/d2f7feffc9665cead386ed48426643a4f08a0519/assignments/assignment2/my_NB_hint.py#L42)

**Comment by sma4396** [comment]
I'm still still struggling to duplicate your output. I'm getting the right classes but I think Im struggling to understand how to deal with the missing values. I've tried many different thinks but my statistics don't quite match the desired output (I'm getting the categories right but not the percentages). When calculating n, v, and nc, am I removing the missing values for that attribute first before calculating all 3? I'm also getting total probability across all classes slightly above 1 sometimes or well below 1. Is this possible?

**Comment by azhe825** [comment]
@sma4396 
1. For simplicity, you do not need to handle missing values specifically (just treat them as a specific category, e.g. "?" as a specific value similar to all values such as 't' or 'f'). I know this is not the perfect way to do that but it does not require users to define a symbol for missing values.
2. The total probability across all classes should be 1, but the floating numbers may not add up to exactly 1 (should be very close though).

**Comment by Goitsemedi888** [comment]
this assignment is proving too challenging given my less than elementary
grasp of python.

**Comment by azhe825** [comment]
@Goitsemedi888 Try the following:
0. Watch the recording for today’s lecture as it explains more about A2 in the end.
1. Reach out to the student with highest Python expertise in your study group for help
2. Reach out to the TA for help
3. Post your specific questions as new issues here.

**Comment by Goitsemedi888** [comment]
Greetings of the day,

this group is not working out as productively as I had hoped. In the past
few days I have become good at regular expressions and searching phrases on
google. I understand the assignment, I know why you set it up in classes so
we can use this model and just make adjustments as the semester progresses.

What you are proposing is that I go to a psychward for mental health help
and consult with the patients.

In one of your lectures you spoke about a machine learning algorithm that
predicts a students overall performance based on previous examples. Machine
learning has its limitations as it cannot account for just being
discouraged and lack of resources for a seemingly insurmountable task.

I have watched my share of youtube videos and am no closer to a
breakthrough on the code.

I will attend the TA's hours maybe theres hope there

**Comment by azhe825** [comment]
@Goitsemedi888 
I am sorry you feel it this way for the assignment. This is the best way in my opinion to learn the concept of machine learning and Python programming. I know it can be difficult for students without a solid programming background but if you stay with the course, you will be better at it in the end. 

Similarly as the previous semester, I will give all of you a chance to make up for the assignment grades towards the end of the semester. Therefore, you just need to try your best for now and keep practicing.
